src/rtcp/receiverReport.ts

```diff
@@ -0,0 +1,342 @@
+	 */
+	getBuffer(): Buffer
+	{
+		return this.buffer;
+	}
+
+	/*
+	 * Get receiver SSRC.
+	 */
+	getSsrc(): number
+	{
+		return this.buffer.readUInt32BE(0);
+	}
+
+	/*
+	 * Set receiver SSRC.
+	 */
+	setSsrc(ssrc: number): void
+	{
+		this.buffer.writeUInt32BE(ssrc, 0);
+	}
+
+	/*
+	 * Get fraction lost.
+	 */
+	getFractionLost(): number
+	{
+		return this.buffer.readUInt8(4);
+	}
+
+	/*
+	 * Set fraction lost.
+	 */
+	setFractionLost(fractionLost: number): void
+	{
+		this.buffer.writeUInt8(fractionLost, 4);
+	}
+
+	/*
+	 * Get total lost.
+	 */
+	getTotalLost(): number
+	{
+		let value = this.buffer.readUIntBE(5, 3);
+
+		// Possitive value.
+		if (((value >> 23) & 1) == 0)
+		{
+			return value;
+		}
+
+		// Negative value.
+		if (value != 0x0800000)
+		{
+			value &= ~(1 << 23);
+		}
+
+		return -value;
+	}
+
+	/*
+	 * Set total lost.
+	 */
+	setTotalLost(totalLost: number): void
+	{
+		// Get the limit value for possitive and negative totalLost.
+		const clamp = (totalLost >= 0) ? totalLost > 0x07FFFFF ? 0x07FFFFF : totalLost
+			: -totalLost > 0x0800000 ? 0x0800000 : -totalLost;
+
+		const value = (totalLost >= 0) ? (clamp & 0x07FFFFF) : (clamp | 0x0800000);
+
+		this.buffer.writeUIntBE(value, 5, 3);
+	}
+
+	/*
+	 * Get highest RTP sequence number.
+	 */
+	getHighestSeqNumber(): number
+	{
+		return this.buffer.readUIntBE(8, 4);
+	}
+
+	/*
+	 * Set highest RTP sequence number.
+	 */
+	setHighestSeqNumber(lastSeq: number): void
+	{
+		this.buffer.writeUIntBE(lastSeq, 8, 4);
+	}
+
+	/*
+	 * Get interarrival jitter.
+	 */
+	getJitter(): number
+	{
+		return this.buffer.readUIntBE(12, 4);
+	}
+
+	/*
+	 * Set interarrival jitter.
+	 */
+	setJitter(jitter: number)
+	{
+		this.buffer.writeUIntBE(jitter, 12, 4);
+	}
+
+	/*
+	 * Set last Sender Report timestamp.
+	 */
+	getLastSRTimestamp(): number
+	{
+		return this.buffer.readUIntBE(16, 4);
+	}
+
+	/*
+	 * Set last Sender Report timestamp.
+	 */
+	setLastSRTimestamp(lsr: number): void
+	{
+		this.buffer.writeUIntBE(lsr, 16, 4);
+	}
+
+	/*
+	 * Get delay since last Sender Report.
+	 */
+	getDelaySinceLastSR(): number
+	{
+		return this.buffer.readUIntBE(20, 4);
+	}
+
+	/*
+	 * Set delay since last Sender Report.
+	 */
+	setDelaySinceLastSR(dlsr: number): void
+	{
+		this.buffer.writeUIntBE(dlsr, 20, 4);
+	}
+}
+
+export class ReceiverReportPacket extends RtcpPacket
+{
+	// Receiver Reports.
+	private reports: ReceiverReport[] = [];
+	// Whether serialization is needed due to modifications.
+	private serializationNeeded: boolean = false;
+
+	// Packet Type.
+	static packetType = PacketType.RR;
+
+	/**
+	 * @param buffer - If given it will be parsed. Otherwise an empty RTP packet
+	 *   will be created.
+	 */
+	constructor(buffer?: Buffer)
+	{
+		super(ReceiverReportPacket.packetType);
+
+		// If no buffer is given, create an empty one with minimum required length.
+		if (!buffer)
+		{
+			this.buffer = Buffer.alloc(FIXED_HEADER_LENGTH);
+
+			return;
+		}
+
+		if (!isRtcp(buffer))
```

Comment:
yes [here](https://github.com/versatica/rtp.js/pull/1/files#diff-ed9aa03064c98f2bfb98b1d4cdbe4df5R36)

Reply:
Yes, I know. I mean that it SHOULD accept passing NO buffer as per the constructor doc.

Reply:
`isRtcp(buffer)` won't be called if no buffer is provided. Constructor would return few lines above.

Reply:
Oh true. However, if the app calls isRtcp() with that packet created without buffer it must return true. Does it?

Reply:
No until `serialize` is called

Reply:
Why? in `RtpPacket` you can do this:

```ts
const packet = new RtpPacket(); // No buffer.

isRtp(packet); // => true
```

I see no reason to allow creating a `RtcpPacket` instance and then having a `isRtcp()` utility that tells us that "it's not a RTCP packet". Also for consistency with `RtpPacket`. Please check the constructor of `RtpPacket` (when no `buffer` is given). It adds the RTP version and stuff for it to be a valid packet since that.